lib/hockey_gerrit.rb

```diff
@@ -1,41 +1,91 @@
 require_relative 'hockey_gerrit/version'
 require 'English'
 
-class HockeyGerrit
-  attr_accessor :output_file, :change, :patch, :log, :log_line
-  def changes(gerrit)
-    raise 'No line from Gerrit provided' if gerrit.empty?
-    gerrit_split = gerrit.split('/')
-    raise 'ENV GERRIT_REFSPEC has invalid format' unless gerrit_split.size >= 2
-    @change, @patch = gerrit_split[-2..-1]
-  end
+# Fix undefined method `command' for main:Object
+# shenzhen depends on commander gem.
+def command(_param)
+end
 
-  def assign_log
-    @log = `git log --reverse -1 --format="%an: %s"`
-    raise 'log command failed' unless $CHILD_STATUS.success?
-  end
+require 'rubygems'
+require 'shenzhen'
+require 'shenzhen/plugins/hockeyapp'
 
-  def write_file
-    File.write(output_file, log_line)
-  end
+module HockeyGerrit
+  class << self
+    def gerrit_refspec(gerrit=ENV['GERRIT_REFSPEC'])
+      raise 'Must set GERRIT_REFSPEC' unless gerrit && !gerrit.empty?
+      gerrit_split = gerrit.split('/')
+      raise 'GERRIT_REFSPEC is invalid' unless gerrit_split.size >= 2
+      change, patch = gerrit_split[-2..-1]
+      "g#{change},#{patch}"
+    end
 
-  def delete_file
-    File.delete(output_file)
-  end
+    def git_log
+      git_log = `git log --reverse -1 --format="%an: %s"`
+      raise 'command failed' unless $CHILD_STATUS.success?
+      git_log
+    end
 
-  def change_line
-    raise 'Change empty' if change.empty?
-    raise 'Patch empty' if patch.empty?
-    raise 'Log empty' if log.empty?
-    @log_line = "g#{change},#{patch}\n#{log}"
-  end
+    def git_commit_sha
+      sha = `git rev-parse --verify HEAD`
+      raise 'command failed' unless $CHILD_STATUS.success?
+      sha
+    end
+
+    HOCKEY_TOKEN = ENV['HOCKEY_TOKEN']
+    IPA = ARGV.first
+    BUILD_URL = ENV['BUILD_URL']
+
+    def run
+      raise 'Must set HOCKEY_TOKEN' unless HOCKEY_TOKEN
+      raise 'Must provide path to IPA' unless IPA
+      raise 'IPA doesn\'t exist' unless File.exist?(IPA)
+      raise 'Must provide BUILD_URL' unless BUILD_URL
+
+      gerrit = gerrit_refspec
+      log = git_log
+      release_notes = "#{gerrit}\n#{log}"
+
+      tries = 1
```

Comment:
This is why we do code reviews! Nice catch. 

Fixed. I had temporarily set it to 1 for debugging.